### DISCUSSION TITLE
Intermittent client hang

### DISCUSSION THREAD

**Discussion by brcisna** [ltsp/ltsp#360] [General]
Hello All,

Dell All-In-One client 16 GB ram X 2 ( about 7 years old) have two of these do the same thing. 

Occasionally on either of these client machines if I have about 12 browser tabs open and about 4-5 multimedia apps open for video editing  the client will simply lose any mouse inputs. The cursor becomes a 'hand' and can not click anything. I can howerver do a CTL-Alt-F2 and get a command propmt login (sometimes) and i see very slowly respawning ,, 'loading rtlwifi direct, r8188eu, rtl8188fw,bin'
The one common thing is the above line always appears at the command prompt login. As an aside this wifi adapter is never actually used on the client machine just for clarification. If i run top at this point the machine is not under any load or memory exhausted.
It simply appears possibly the realtek wifi firmware may be hanging the machine.
There is nothing obvious in dmesg that would give me any clues. 
Have let it set for a few minutes just to see if it would ever come back,but never does.
It is very rare for this to happen but in time it will.

Just curious if anyone has ever experienced this scenario on a  client machine.?

Thanks

**Comment by alkisg** [comment]
Hello Barry,

1. Try to blacklist the module (or unload it after boot with `modprobe -r modulename`), so that you can see if it is to blame or not
2. When the problem happens, can you still see the client in Epoptes?

**Comment by brcisna** [comment]
Hello All,

Disabled what thought may have been a module causing the client hang ,and further investigation am fairly certain found what is causing the client random hang.
Never noticed all the 'tracker- extract & tracker-miner' process running on client and ltsp server until a few months ago,but may have been there even when installed about one year ago,and just never noticed these running. Also the system log is constantly full of tracker items.

Found this:
https://gitlab.gnome.org/GNOME/tracker/-/issues/95

running 'tracker status' yielded this:
tsp110:~$ tracker status
Currently indexed: 58074 files, 2861 folders
Remaining space on database partition: 10.5 TB (89.57%)
Data is still being indexed: Estimated 01h 26m 01s left

So i done this:
https://www.linuxuprising.com/2019/07/how-to-completely-disable-tracker.html

Since disabling the tracker running,have not had a hang on client.

Aslo looking back,i noticed this hang seemed to always happen if i was clicking around in PCmanFM very quickly manipulting video files for editing, so this tracker thing kind of makes sense.
My findings are usually not very scientific, but think this may in fact have been the problem.

I will continue to monitor this and will eventually close this discussion. 

EDIT: My client after disabling the r8188eu module,but would still hang if clicking files very quickly in PCmanFM, would freeze but the mouse could still be run around. I could alos do a CTL-ALT-F2 and at the commandline login kill various processes,and also could ping the ltsp server. It was as if the screen was frozen with whatever image was showing at the time of lockup. Even after closing the processes running they never sdisappeared from the screen. Kind of hard to explain without seeing this.  Also as mentioned in the top most link,I have tons of multimedia files as did the user experiencing the same porblem,,only mine was slightly different hang.

Thanks
